project/src/main/java/com/google/autograder/data/Database.java

```diff
@@ -130,13 +144,95 @@ public String getAllQuestionsJSON(String key) {
             int points = Math.toIntExact(pointsLong);
             String status = (String) entity.getProperty("status");
             String assignmentKey = (String) entity.getProperty("assignmentKey");
-            Question currQuestion = new Question(name, points, status, assignmentKey);
+            Key questionKey = entity.getKey();
+            Question currQuestion = new Question(name, points, status, assignmentKey, questionKey);
             questions.add(currQuestion);
         }
         String json = new Gson().toJson(questions);
         return json;
     }
-    return new Gson().toJson("");
+    catch (Exception e) {
+        System.out.println("assignment wasn't found");
```

Comment:
could you also print the stack trace of the exception using

e.printStackTrace(System.out);

Reply:
yes